vital/plugin_loader/plugin_manager.cxx

```diff
@@ -72,7 +74,18 @@ class plugin_manager::priv
     : m_all_loaded( false )
     , m_loader( new plugin_loader( register_function_name, shared_library_suffix ) )
     , m_logger( kwiver::vital::get_logger( "vital.plugin_manager" ) )
-  { }
+  {
+    using kvpf = kwiver::vital::plugin_factory;
+
+    // Add the default filter which checks for duplicate plugins
+    plugin_filter_handle_t filt = std::make_shared<kwiver::vital::plugin_filter_default>();
+    m_loader->add_filter( filt );
+
+    // Add filter that excludes applets
+    filt = std::make_shared<kwiver::vital::plugin_filter_category>(
```

Comment:
It seems a bit odd to me that the default behavior is to block applets, and the only way to undo this is to clear all filters and then manually recreate the default one.  I'm not convinced that applets should have this special status. There are many applications that don't want sprokit processes either, but that is not off by default.

It doesn't need to hold up this branch, but it would be nice if there was a high-level way for end users to specify which categories they want and have the filters used internally to achieve that effect.  For example, 
```c++
auto& vpm = kwiver::vital::plugin_manager::instance();
vpm.block_plugin_category(kvpf::APPLET_CATEGORY);
```
and that helper function would create the filter for you.  Then we could leave Applet out of the defaults and let users block it when needed.  Alternatively this could be opt-in where everything is blocked by default and the user lists which things they want to enable.

Reply:
I guess I didn't look that closely at the implementation either, aside from being surprised that applets still get loaded at all.

I've talked to Linus and there are some thoughts for further refinements. In particular, I would like to see plugins sorted into subdirectories according to classification, such that the plugin manager never even touches plugins outside the subdirectories of the requested types. I would also prefer to see the types be constrained to a fixed set of values (we would possibly want an open-ended "utility" or some such to catch anything not in the other categories) and a bitmask, passed to `load_all_plugins`, used to specify what types are desired.